### DISCUSSION TITLE
getStaticPaths ran but getStaticProps didn't

### DISCUSSION THREAD

**Discussion by Senbonzakura1234** [vercel/next.js#52537] [Help] [answered]
### Summary

```
"next": "13.4.8",
"react": "18.2.0",
"@trpc/client": "^10.33.1",
"@trpc/next": "^10.33.1",
"@trpc/react-query": "^10.33.1",
"@trpc/server": "^10.33.1",
```

I was trying to implement SSG for my effect detail page but it didn't work. On run build the `getStaticPaths` seems run fine and return all the effect Id as expected. But it seems `getStaticProps` didn't run and the id in the component was undefined

I was following [trpc SSG guide](https://trpc.io/docs/client/nextjs/ssg) and [NextJs SSG guide](https://nextjs.org/docs/pages/building-your-application/rendering/static-site-generation)

in my `src\pages\effect\[id].tsx`

```typescript  
import { appRouter } from '@root/server/api/router/_app';
import { db } from '@root/server/db';
import { effects } from '@root/server/db/schema';
import { apiContext } from '@root/utils/trpc';
import { createServerSideHelpers } from '@trpc/react-query/server';
import type { GetStaticPaths, GetStaticPropsContext, InferGetStaticPropsType } from 'next';

export const getStaticPaths: GetStaticPaths = async () => {
  const effectIds = await db.select({ id: effects.id }).from(effects);

  return {
    paths: effectIds.map(({ id }) => {
      console.log(id);
      return { params: { id } };
    }),
    fallback: 'blocking',
  };
};

export const getStaticProps = async (context: GetStaticPropsContext<{ id: string }>) => {
  const helpers = createServerSideHelpers({ router: appRouter, ctx: {} });
  const id = context.params?.id;
  // the code didn't throw at this point
  if (!id) throw new Error('Debugging - id is undefined - getStaticProps');

  await helpers.effect.getOne.prefetch({ id });
  return { props: { trpcState: helpers.dehydrate(), id }, revalidate: 1 };
};

export default function EffectDetail({ id }: InferGetStaticPropsType<typeof getStaticProps>) {
  // the code throw at this point, means `getStaticProps` didn't run at all
  if (!id) throw new Error('Debugging - id is undefined - client');

  const { data, isError, error } = apiContext.effect.getOne.useQuery({ id });

  return (
    // render Effect ...
  );
}
```

### Additional information

```js
full code implement https://github.com/Senbonzakura1234/sophie-dex/tree/feature/implement-ssg
```

**Comment by icyJoseph** [comment]
Are you sure `getStaticProps` is not running? could place some logs on it?

At first I thought that maybe `_app` was not passing along pageProps, but it does...

I was also thinking that `fallback` could be the issue, but you have it as `blocking`, so a fallback page is not needed... 🤔 

Look at the EffectDetail entire props, is it empty?

```tsx
export default function EffectDetail({ id, ...rest }: InferGetStaticPropsType<typeof getStaticProps>) {
   console.log(rest, {id})
```

**Reply by Senbonzakura1234** [reply]
@icyJoseph no it didn't, tried adding `console.log('getStaticProps ran');` at the top but no log shows up

**Reply by Senbonzakura1234** [reply]
and the props did be empty, no props were transfered to the component
![image](https://github.com/vercel/next.js/assets/43206347/c419769f-66a4-4575-905b-d19650301b29)

**Reply by icyJoseph** [reply]
Oh no, that's "no bueno" ~

Could you please, do a `console.log(useRouter().isFallback)` log in the component as well?

There can be a number of issues here... so let's start to unravel them one by one...

What's the typeof `id`? Is it an ObjectId kind of thing that resolves to string, but is really an object? The pre-rendering message seems to suggest it is a plain string...

**Reply by Senbonzakura1234** [reply]
@icyJoseph Its value is `false`
![image](https://github.com/vercel/next.js/assets/43206347/4a0f5109-04ef-4f80-a27d-5a7d361827d5)

btw forgot to mention my packages version:
```
"next": "13.4.8",
"react": "18.2.0",
"@trpc/client": "^10.33.1",
"@trpc/next": "^10.33.1",
"@trpc/react-query": "^10.33.1",
"@trpc/server": "^10.33.1",
```

**Reply by Senbonzakura1234** [reply]
@icyJoseph  about the id, it was just pure string, the format was uuid though

**Reply by icyJoseph** [reply]
If you were to remove the tRCP parts, and with that any data fetching, just to make sure that things work with "vanilla" next.js - so that you end up seeing the `id` on the Effect page.

**Reply by Senbonzakura1234** [reply]
@icyJoseph okay i tried remove all trpc queries and it still face the same error. But after that i also remove the `withTRPC` in '_app.tsx' and suddenly it works now. Maybe the problem lies in the trpc context or trpc router

**Reply by icyJoseph** [reply]
yeah I suspect that too, but not sure why they are messing with `pageProps`. Also, why is not GSP running? apparently? Could you open a discussion on their GitHub Page?

**Comment by Senbonzakura1234** [comment, marked as answer]
@icyJoseph Okay so i found the reason why. I set `ssr` option to `true` in `createTRPCNext`, it has to set to `false`.
